src/bonuses.cpp

```diff
@@ -0,0 +1,272 @@
+            std::make_pair( "speed", AFFECTED_SPEED ),
+            std::make_pair( "movecost", AFFECTED_MOVE_COST ),
+            std::make_pair( "damage", AFFECTED_DAMAGE ),
+            std::make_pair( "arm", AFFECTED_ARMOR ),
+            std::make_pair( "arpen", AFFECTED_ARMOR_PENETRATION )
+        }
+    };
+    const auto &iter = stat_map.find( s );
+    if( iter != stat_map.end() ) {
+        return iter->second;
+    }
+
+    return AFFECTED_NULL;
+}
+
+bonus_container::bonus_container()
+{
+}
+
+// Duplicates cut bonuses into stab bonuses
+void bonus_container::legacy_fixup( bonus_map &bm )
+{
+    bool changed_something = false;
+    do {
+        for( const auto &pr : bm ) {
+            if( pr.first.get_damage_type() != DT_CUT ) {
+                continue;
+            }
+
+            const affected_type as( pr.first.get_stat(), DT_STAB );
+            if( bm.count( as ) != 0 ) {
+                continue;
+            }
+
+            changed_something = true;
+            const auto value_copy = pr.second;
+            bm[as] = value_copy;
+            // Iterators are now scrambled, we need to restart the loop
+            break;
+        }
+    } while( changed_something );
+}
+
+void bonus_container::load( JsonObject &jo )
+{
+    // Warning: ugly string hackery
+    static const std::string mult( "mult" );
+    const auto &members = jo.get_member_names();
+    for( const auto &member : members ) {
+        const std::vector<std::string> tokens = split( member, '_' );
+        if( tokens.size() < 2 ) {
+            continue;
+        }
+
+        affected_stat as = AFFECTED_NULL;
+        scaling_stat ss = STAT_NULL;
+        damage_type dt = DT_NULL;
+        bool is_mult = false;
+        bool skipped_token = false;
+        for( const auto &token : tokens ) {
+            if( token == mult ) {
+                is_mult = true;
+                continue;
+            }
+
+            damage_type ndt = dt_by_name( token );
+            if( ndt != DT_NULL ) {
+                if( dt != DT_NULL ) {
+                    jo.throw_error( "Damage type set twice", member );
+                }
+                dt = ndt;
+                continue;
+            }
+
+            affected_stat nas = affected_stat_from_string( token );
+            if( nas != AFFECTED_NULL ) {
+                if( as != AFFECTED_NULL ) {
+                    jo.throw_error( "Affected stat set twice", member );
+                }
+                as = nas;
+                continue;
+            }
+
+            scaling_stat nss = scaling_stat_from_string( token );
+            if( nss != STAT_NULL ) {
+                if( ss != STAT_NULL ) {
+                    jo.throw_error( "Scaling stat set twice", member );
+                }
+                ss = nss;
+                continue;
+            }
+
+            skipped_token = true;
+            break;
+        }
+
+        if( skipped_token ) {
+            // This isn't a set of ma_ modifiers
```

Comment:
We need to distinct two different conditions:
1) parameter is not a bonus but it's used in some other place;
2) parameter is not a bonus due to typo in json definition and this parameter isn't used at all.

Former case should not appear in debug log but  latter one surely should. And that's why I'd prefer to keep showing color data warnings (and fix the underlying issue sometime): color generator doing incorrect assumptions about color names and log highlights this fact. Sorry I was unclear in my initial message.